x-pack/filebeat/tests/integration/input_reload_test.go

```diff
@@ -0,0 +1,336 @@
+					t.Fatalf("Unit '%s' is not healthy, state: %s", unit.GetId(), unit.GetState().String())
+				}
+			}
+			return &proto.CheckinExpected{
+				Units: units[idx],
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error { return nil },
+		// ActionsChan: make(chan *mock.PerformAction, 100),
+	}
+
+	require.NoError(t, server.Start())
+	t.Cleanup(server.Stop)
+
+	filebeat := NewBeat(
+		t,
+		"../../filebeat.test",
+		tempDir,
+		"-E", fmt.Sprintf(`management.insecure_grpc_url_for_testing="localhost:%d"`, server.Port),
+		"-E", "management.enabled=true",
+	)
+
+	filebeat.Start()
+
+	// waitDeadlineOr5Mins looks at the test deadline
+	// and returns a reasonable value of waiting for a
+	// condition to be met. The possible values are:
+	// - if no test deadline is set, return 5 minuets
+	// - if a deadline is set and there is less than
+	//   0.5 second left, return the time left
+	// - otherwise return the time left minus 0.5 second.
+	waitDeadlineOr5Min := func() time.Duration {
+		deadline, deadileSet := t.Deadline()
+		if deadileSet {
+			left := deadline.Sub(time.Now())
+			final := left - 500*time.Millisecond
+			if final <= 0 {
+				return left
+			}
+			return final
+		}
+		return 5 * time.Minute
+	}
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("Can only start an input when all related states are finished")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'Can only start an input when all related states are finished' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("file 'flog.log' is not finished, will retry starting the input soon")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'file 'flog.log' is not finished, will retry starting the input soon' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("ForceReload set to TRUE")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'ForceReload set to TRUE' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("Reloading Beats inputs because forceReload is true")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'Reloading Beats inputs because forceReload is true' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("ForceReload set to FALSE")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'ForceReload set to FALSE' not found on Filebeat logs")
+
+	// Set it to true, so the temporaty directory is removed
+	testSucceeded = true
+}
+
+func requireNewStruct(t *testing.T, v map[string]interface{}) *structpb.Struct {
+	str, err := structpb.NewStruct(v)
+	if err != nil {
+		require.NoError(t, err)
+	}
+	return str
+}
+
+// generateLogFile generates a log file by appending the current
+// time to it every second.
+func generateLogFile(t *testing.T, fullPath string) {
```

Comment:
Can we at least accept a context as an argument so we can stop the go routine if we want to?

Reply:
We should not leak that, thanks!

I'll implementing something using `t.Cleanup` so the caller does not need to worry about any cleanup.

Reply:
Done on 49f314d79f.